src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/InstanceRepository.cs

```diff
@@ -444,20 +444,23 @@ public async Task<List<Instance>> GetInstancesInStateOfInstanceOwner(int instanc
             {
                 filter = _client.CreateDocumentQuery<Instance>(_collectionUri, feedOptions)
                         .Where(i => i.InstanceOwnerId == instanceOwnerIdString)
+                        .Where(i => (!i.VisibleDateTime.HasValue || i.VisibleDateTime < DateTime.UtcNow))
                         .Where(i => !i.InstanceState.IsDeleted)
                         .Where(i => !i.InstanceState.IsArchived);
             }
             else if (instanceState.Equals("deleted"))
             {
-                // what about hard delete. Should we account for that too?
                 filter = _client.CreateDocumentQuery<Instance>(_collectionUri, feedOptions)
                         .Where(i => i.InstanceOwnerId == instanceOwnerIdString)
-                        .Where(i => i.InstanceState.IsDeleted);
+                        .Where(i => (!i.VisibleDateTime.HasValue || i.VisibleDateTime < DateTime.UtcNow))
+                        .Where(i => i.InstanceState.IsDeleted)
```

Comment:
In which scenario do you think that the visibledatetime must be considered for archived and delete?

Reply:
Dunno, but I figure it can't hurt to have the filter there too?